### PR TITLE
Feature/initial framework code

### PR DESCRIPTION
**Pull request by drhaynes**
This sets up circle to build the framework.

### CODE REVIEW

**Comment by dhardiman** [comment]
Is this repo still private? I notice it depends on grid point, which definitely still is. Maybe we should prioritise publishing that on Monday.

File: .gitignore [hunk 1]
```diff
@@ -63,3 +63,5 @@ fastlane/report.xml
 fastlane/screenshots
 APIKEY
 
+# WhirlyGlobe-Maply and its associated submodules are checked out here
+Submodules/WhirlyGlobe
```
Comment (dhardiman) [inline .gitignore]: Probably don't need to ignore this. Git submodules will understand it well enough...

File: Carthage/Checkouts/osgridpointconversion-ios [hunk 2]
```diff
@@ -0,0 +1 @@
+Subproject commit 37f3937ae41dcebc35f6ec5f9cfbc68ee3ec25b2
```
Comment (dhardiman) [inline Carthage/Checkouts/osgridpointconversion-ios]: This can possibly go though?

File: MaplyAdapter/OSBNGUtils.m [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+//
+
+#import "OSBNGUtils.h"
+@import OSGridPointConversion;
+#import "MaplyComponent.h"
+
+@implementation OSBNGUtils
+
++ (MaplyProj4CoordSystem *)bngCoordinateSystem {
+    NSString *proj4Str = [OSBNGTransformation proj4String];
```
Comment (dhardiman) [inline MaplyAdapter/OSBNGUtils.m]: As with today's adventures, probably want the non-OSTN02 version here for the time being?

File: MaplyAdapter/OSTileSourceFactory.m [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    MaplyQuadImageTilesLayer *layer = [[MaplyQuadImageTilesLayer alloc] initWithCoordSystem:tileSource.coordSys tileSource:tileSource];
+    layer.handleEdges = false;
+    layer.coverPoles = false;
+    layer.drawPriority = kMaplyImageLayerDrawPriorityDefault;
+    layer.numSimultaneousFetches = 8;
+    // Normally we'd try to match the screen resolution with tiles, but that seems
+    // to overfetch OS Maps, so we'll back off a bit.  Smaller number for even more backing off
+    layer.importanceScale = 0.5;
+    layer.singleLevelLoading = true;
+    layer.multiLevelLoads = @[ @(-3) ];
```
Comment (dhardiman) [inline MaplyAdapter/OSTileSourceFactory.m]: Minor stylistic, this could be `@-3`

File: .gitignore (hunk 1 above)
Comment (drhaynes) [inline .gitignore]: Ah didn't realise that. Removed.

File: Carthage/Checkouts/osgridpointconversion-ios (hunk 2 above)
Comment (drhaynes) [inline Carthage/Checkouts/osgridpointconversion-ios]: Yep, thanks. Removed.

File: MaplyAdapter/OSBNGUtils.m (hunk 3 above)
Comment (drhaynes) [inline MaplyAdapter/OSBNGUtils.m]: Good point, will update.

**Comment by drhaynes** [comment]
Making grid point public early next week sounds like a good plan.

**Comment by dhardiman** [comment]
Will merge pending making stuff public next week

**Comment by dhardiman** [comment]
No I won't. "Pull requests that have a failing status can't be merged from a phone."  Wut. Happy you merge if...

**Comment by drhaynes** [comment]
Obviously to prevent drunk merging.

**Pull request merged**